compiler/src/dmd/iasm.d

```diff
@@ -35,7 +35,7 @@ else
 
 /************************ AsmStatement ***************************************/
 
-extern(C++) Statement asmSemantic(AsmStatement s, Scope *sc)
+extern(C++) Statement asmSemantic(AsmStatement s, Scope *sc) nothrow
```

Comment:
You'll need to compile in `iasmgcc.d` instead of `iasmdmd.d`, as I'm sure there's be more nothrow violations in that module (used by GDC and LDC).